example/spec/scaling/scalingSpec.js

```diff
@@ -0,0 +1,115 @@
+let clusterArn;
+let autoScalingGroupName;
+
+async function getClusterStats({
+  statTypes = ['runningEC2TasksCount', 'pendingEC2TasksCount']
+}) {
+  const stats = (await ecs().describeClusters({
+    clusters: [clusterArn],
+    include: ['STATISTICS']
+  }).promise()).clusters[0].statistics;
+  const returnedStats = {};
+  statTypes.forEach((statType) => {
+    returnedStats[statType] = parseInt(find(stats, ['name', statType]).value, 10);
+  });
+  return returnedStats;
+}
+
+async function getAutoScalingGroupName(stackName) {
+  const autoScalingGroups = (await autoscaling().describeAutoScalingGroups({}).promise()).AutoScalingGroups;
+  const asg = find(autoScalingGroups, (group) => group.AutoScalingGroupName.match(new RegExp(stackName, 'g')));
+  return asg.AutoScalingGroupName;
+}
+
+const waitPeriod = 30000;
+async function getNewScalingActivity() {
+  const params = {
+    AutoScalingGroupName: autoScalingGroupName,
+    MaxRecords: 1
+  };
+  let activties = await autoscaling().describeScalingActivities(params).promise();
+  const startingActivity = activties.Activities[0];
+  let mostRecentActivity = Object.assign({}, startingActivity);
+  /* eslint-disable no-await-in-loop */
+  while (startingActivity.ActivityId === mostRecentActivity.ActivityId) {
+    activties = await autoscaling().describeScalingActivities(params).promise();
+    mostRecentActivity = activties.Activities[0];
+    console.log(`No new activity found. Sleeping for ${waitPeriod / 1000} seconds.`);
+    await sleep(waitPeriod);
+  }
+  /* eslint-enable no-await-in-loop */
+
+  return mostRecentActivity;
+}
+
+const numExecutions = 3;
+const numActivityTasks = 1;
+const minInstancesCount = 1;
+
+describe('When a task is configured to run in Docker', () => {
+  beforeAll(async () => {
+    clusterArn = await getClusterArn(config.stackName);
+    autoScalingGroupName = await getAutoScalingGroupName(config.stackName);
+  });
+
+  describe('the load on the system exceeds that which its resources can handle', () => {
+    let workflowExecutionArns = [];
+
+    beforeAll(async () => {
+      const workflowExecutionPromises = [];
+      for (let i = 0; i < numExecutions; i += 1) {
+        workflowExecutionPromises.push(buildAndStartWorkflow(
+          config.stackName,
+          config.bucket,
+          workflowName
+        ));
+      }
+      workflowExecutionArns = await Promise.all(workflowExecutionPromises);
+    });
+
+    afterAll(async () => {
+      const completions = workflowExecutionArns.map((executionArn) => waitForCompletedExecution(executionArn));
+      await Promise.all(completions);
+    });
+
+    it('can handle the load', async () => {
+      await sleep(5000);
+      const stats = await getClusterStats({});
+      expect(stats.runningEC2TasksCount + stats.pendingEC2TasksCount).toEqual(numExecutions + numActivityTasks);
+    });
+
+    it('adds new resources', async () => {
+      console.log('Waiting for scale out policy to take affect.');
```

Comment:
Should this say "Waiting for the scale in policy.."?

Reply:
```suggestion
      console.log('Waiting for scale in policy to take affect.');
```

Reply:
I think "scale out" means "add more EC2 instances" and "scale in" means "remove existing EC2 instances". Is that different from your understanding?

Reply:
I would have said the opposite of that

Reply:
I was starting to feel crazy that I was the only one who speaks this way, but I verified to "scale out" in scaling parlance is to add nodes in a cluster (i.e. to scale out is to expand)

https://en.wikipedia.org/wiki/Scalability
>To scale horizontally (or scale out/in) means to add more nodes to (or remove nodes from) a system, such as adding a new computer to a distributed software application. 

and on [AWS docs](https://docs.aws.amazon.com/autoscaling/ec2/userguide/AutoScalingGroupLifecycle.html#as-lifecycle-scale-out)
>The following scale out events direct the Auto Scaling group to launch EC2 instances and attach them to the group: